nrf_modem/include/nrf_socket.h

```diff
@@ -17,7 +17,22 @@
 extern "C" {
 #endif
 
+#if !defined(__GNUC__) || (__GNUC__ == 0)
```

Comment:
Why check for `__GNUC__ == 0` ?
If we only need to check for the C compiler, we don't need to check that `__GNUC__` is equal to zero.

Reply:
Okey. I basically took this part of code from the history of nrf_modem (v1.5.1). I agree that the !defined part would be enough.